### PR TITLE
Hackathon changes

### PR DESCRIPTION
**Pull request by coszio**
## Summer Camp hackathon + last week of MLH

This last week was the most intense of the program, and as a team we managed to get all of the following changes done:

- feat: login component  
- content: replace static content  
- integration: integrate donations
- refactor: cleanup fields in grants-program  
- feat: donation flow
- integration: fetch grants for explore view 
- integration: fetch grants for detail view 
- feat: support author (directly)  
- feat: create grant flow
- ui: markdown editor in create grant 
- ui: used toastify instead of notify in getgrants
- ui: add loading spinner in normal donation popup 
- ui: add default grant image fallback in getgrants

Thanks @RishiDesai17 @yashikajotwani12 @cdhiraj40 @Mayank17M for coming in _clutch_ with these.

### Note
All other PRs are basically included in this one, so we'll close them in favor of this.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (jacobcreech) [inline .vscode/launch.json:1]: Do we need this file? Very specific to local setup

File: app/src/transactions/index.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
-import * as anchor from "@project-serum/anchor";
-import { Program } from "@project-serum/anchor";
-import { GrantsProgram } from "../idl/grants_program";
+// import * as anchor from "@project-serum/anchor";
```
Comment (jacobcreech) [inline app/src/transactions/index.ts:1]: Please clean up comments

File: app/src/transactions/makeDonation.ts [hunk 3]
```diff
@@ -1,15 +1,29 @@
 import * as anchor from "@project-serum/anchor";
-import { BN } from "@project-serum/anchor";
+import { BN, Program } from "@project-serum/anchor";
 import { encode } from "@project-serum/anchor/dist/cjs/utils/bytes/utf8";
 import { PublicKey } from "@solana/web3.js";
 import { toBytesInt32 } from "../utils/conversion";
-import { program } from "./index";
+import { GrantsProgram } from "../idl/grants_program";
+
+// let program: Program<GrantsProgram> = undefined;
```
Comment (jacobcreech) [inline app/src/transactions/makeDonation.ts:8]: Please clean up comments

**Review by jacobcreech — CHANGES_REQUESTED** [review]
Few nits. Looking good!

File: .vscode/launch.json (hunk 1 above)
Comment (coszio) [reply .vscode/launch.json:1]: True, I'll add the `.vscode` folder to `.gitignore`

**Review by jacobcreech — APPROVED** [review]

**Pull request merged**